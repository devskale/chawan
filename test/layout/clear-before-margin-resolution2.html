<!DOCTYPE html>
<div style="display: flow-root; background: red; width: 20ch">
<div style="margin-top: 1em">
<div style="float: right">right</div>
<div style="clear: both; margin-top: 3em"></div>
<div style="margin-top: 2em">
test
</div>
</div>
</div>

<div style="display: flow-root; background: gainsboro; width: 20ch">
<div style="margin-top: 1em">
<div style="float: right">right</div>
<div style="clear: both; margin-top: 3em; display: flow-root"></div>
<div style="margin-top: 2em">
test
</div>
</div>
</div>

<div style="display: flow-root; background: blue; width: 20ch">
<div style="margin-top: 1em">
<div style="float: right; height: 2em">right</div>
<div style="clear: both; margin-top: 2em"></div>
<div style="margin-top: 3em">
test
</div>
</div>
</div>

<div style="display: flow-root; background: purple; width: 20ch">
<div style="margin-top: 1em">
<div style="float: right">right</div>
<div style="clear: both; margin-top: -2em"></div>
<div style="margin-top: 3em">
test
</div>
</div>
</div>

<div style="display: flow-root; background: yellow; width: 20ch">
<div style="margin-top: 1em">
	<div style="float: right">test</div>
	<div style="clear: left; margin-top: 2em"></div>
		<div style="margin-top: 3em">
		test
		</div>
	</div>
</div>

<div style="display: flow-root; background: brown; width: 20ch">
	<div style="margin-top: 1em">
		<div style="float: left">test</div>
		<div style="float: right">test</div>
		<div style="clear: left; margin-top: 2em"></div>
		<div style="margin-top: 3em">
		test
		</div>
	</div>
</div>

<div style="display: flow-root; background: orange; width: 20ch">
	<div style="margin-top: 1em">
	<div style="margin-top: 2em"></div>
		<div style="margin-top: 3em">
		test
		</div>
	</div>
</div>

<div style="margin-top: 10em; clear: both">
????
</div>

<div style="display: flow-root; background: red; width: 20ch">
<div style="margin-top: 1em; background: blue">
<div style="float: right">right</div>
<div style="clear: both; margin-top: 1em"></div>
<div style="margin-top: 2em">
test
</div>
</div>
</div>
