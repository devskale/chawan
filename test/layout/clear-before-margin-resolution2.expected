[48;2;255;0;0m                    [49m
[48;2;255;0;0m               right[49m
[48;2;255;0;0mtest                [49m
[38;2;104;104;104m[48;2;220;220;220m                    [39m[49m
[38;2;104;104;104m[48;2;220;220;220m               right[39m[49m
[38;2;104;104;104m[48;2;220;220;220m                    [39m[49m
[38;2;104;104;104m[48;2;220;220;220m                    [39m[49m
[38;2;104;104;104m[48;2;220;220;220mtest                [39m[49m
[48;2;0;0;255m                    [49m
[48;2;0;0;255m               right[49m
[48;2;0;0;255m                    [49m
[48;2;0;0;255m                    [49m
[48;2;0;0;255mtest                [49m
[48;2;128;0;128m                    [49m
[48;2;128;0;128m               right[49m
[48;2;128;0;128m                    [49m
[48;2;128;0;128m                    [49m
[48;2;128;0;128m                    [49m
[48;2;128;0;128mtest                [49m
[38;2;109;109;109m[48;2;255;255;0m                    [39m[49m
[38;2;109;109;109m[48;2;255;255;0m                    [39m[49m
[38;2;109;109;109m[48;2;255;255;0m                    [39m[49m
[38;2;109;109;109m[48;2;255;255;0mtest            test[39m[49m
[48;2;165;42;42m                    [49m
[48;2;165;42;42mtest            test[49m
[48;2;165;42;42m                    [49m
[48;2;165;42;42mtest                [49m
[38;2;57;57;57m[48;2;255;165;0m                    [39m[49m
[38;2;57;57;57m[48;2;255;165;0m                    [39m[49m
[38;2;57;57;57m[48;2;255;165;0m                    [39m[49m
[38;2;57;57;57m[48;2;255;165;0mtest                [39m[49m










????
[48;2;255;0;0m                    [49m
[48;2;0;0;255m               right[49m
[48;2;0;0;255m                    [49m
[48;2;0;0;255mtest                [49m
